### PR TITLE
Initial integration tests for ec2_vpc_dhcp_option

### PR DESCRIPTION
**Pull request by jillr**
##### SUMMARY
Based on work started by s-hertel in #109

##### ISSUE TYPE
- New Tests Pull Request

##### COMPONENT NAME
ec2_vpc_dhcp_option

### CODE REVIEW

**Comment by jillr** [comment]
@s-hertel I opened a new PR for this one so you don't have to keep track of merging my changes into your branch as we iterate, let me know if you'd prefer to keep this work in 109!

File: tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,897 @@
...
+      that:
+        - dhcp_options.changed
+        - dhcp_options.new_options
+        - dhcp_options.new_options.keys() | list | sort == ['domain-name', 'domain-name-servers', 'netbios-name-servers', 'netbios-node-type', 'ntp-servers']
+        - dhcp_options.new_options['ntp-servers'] | sort == ['10.0.0.2', '10.0.1.2']
+        - dhcp_options.new_options['netbios-name-servers'] | sort == ['10.0.0.1', '10.0.1.1']
+        # FIXME: module/aws randomly returns as a string or list
+        - dhcp_options.new_options['netbios-node-type'] in ['2', ['2']]
+        # found I think is false, because the options are different and no id is provided?
+        - dhcp_options.new_options['domain-name'] == "{{ aws_domain_name }}"
```
Comment (goneri) [inline tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml:185]: `dhcp_options.new_options['domain-name']` is a list. This is probably the reason why the test fail here. Is this expected?

Comment (goneri) [reply tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml:185]: cf. https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/ec2_vpc_dhcp_option.py#L325

Comment (jillr) [reply tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml:185]: It depends. :)  Thanks, I thought I caught all of these.

**Comment by jillr** [comment]
Something is intermittently misbehaving when setting tags, I haven't been able to reproduce it locally yet but it's causing the unstable tests. Will have to dig into this more.

**Comment by tremble** [comment]
@jillr Given that this is likely the module being unstable rather than the tests being flaky I'd suggest we merge these tests so we have tests for the boto2 based behaviour, and look to move this module over to boto3.

Alternatively you may find that if you pass the ID of the options when setting the tags it's more stable than a 'search' based run.

File: tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,893 @@
...
+      delete_old: True
+      ntp_servers:
+          - 10.0.0.2
+          - 10.0.1.2
+      netbios_name_servers:
+          - 10.0.0.1
+          - 10.0.1.1
+      tags:
+        AnsibleTest: integration
+        Collection: amazon.aws
```
Comment (tremble) [inline tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml:735]: We have a history of accidentally mangling tags (during snake-/Camel- case conversions) in modules.  I'd like to see some additional tag based tests here.

```
tags:
  "lowercase spaced": 'hello cruel world'
  "Title Case": 'Hello Cruel World'
  CamelCase: 'SimpleCamelCase'
  snake_case: 'simple_snake_case'
```

Comment (jillr) [reply tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml:735]: I'm not sure about the "Title Case" test, I couldn't find other examples of where we're doing this, but it's in there now. What do you think?

Comment (tremble) [reply tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml:735]: Yeah that was me try to be thorough on the test suite I'd started writing.  I think we need to look at being more consistent in general with out tagging tests.  The most common issues are with the other 3, but I don't think adding extra tags would hurt.

Comment (jillr) [reply tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml:735]: Sorry not opposed to having more tests. More that I wasn't sure `['title _case']` was a desirable result, though it is what the module is doing.

File: tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,930 @@
...
+      that:
+        - dhcp_options.changed
+        - not dhcp_options.new_options
+        - new_dhcp_options_id == dhcp_options.dhcp_options_id
+        - dhcp_options_info.dhcp_options[0].tags is defined
+        - dhcp_options_info.dhcp_options[0].tags.keys() | length == 4
+        - dhcp_options_info.dhcp_options[0].tags['lowercase spaced'] == 'hello cruel world'
+        - dhcp_options_info.dhcp_options[0].tags['title _case'] == 'Hello Cruel World'
+        - dhcp_options_info.dhcp_options[0].tags['camel_case'] == 'SimpleCamelCase'
+        - dhcp_options_info.dhcp_options[0].tags['snake_case'] == 'simple_snake_case'
```
Comment (tremble) [inline tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml:788]: The middle 2 are bugs: it should be coming back as 
```suggestion
        - dhcp_options_info.dhcp_options[0].tags['Title Case'] == 'Hello Cruel World'
        - dhcp_options_info.dhcp_options[0].tags['CamelCase'] == 'SimpleCamelCase'
        - dhcp_options_info.dhcp_options[0].tags['snake_case'] == 'simple_snake_case'
```

I guess we comment them out and start working on a v3 implementation with a pile of bug fixes...

File: tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml (hunk 2 above)
Comment (tremble) [reply tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml:735]: It's not a desirable result, which is why I wanted to test it :)

**Review by tremble — APPROVED** [review]
I think we're good with these as a starting point.  My eyes are glazing over so I'd appreciate if @goneri could take another look too.

**Review by goneri — APPROVED** [review]
Works for me, I will address the tag problem in another PR.

**Pull request merged**